### PR TITLE
Feature/add security

### PR DESCRIPTION
**Pull request by locpnguyen2912**

### CODE REVIEW

File: src/main/java/com/example/restapi/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                .antMatchers(HttpMethod.PUT, "/employees/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.DELETE, "/employees/**").hasRole("ADMIN")
+                .and()
+                .csrf().disable()
+                .formLogin().disable();
+    }
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
+        PasswordEncoder encoder =
```
Comment (vokhanhhuy) [inline src/main/java/com/example/restapi/config/SecurityConfig.java]: what if I want to use `BCryptPasswordEncoder` as my password encoder? Is it still working?

File: src/main/java/com/example/restapi/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                PasswordEncoderFactories.createDelegatingPasswordEncoder();
```
Comment (vokhanhhuy) [inline src/main/java/com/example/restapi/config/SecurityConfig.java]: Why are we encoding the password while we put password as plain text in API request call?

File: src/main/java/com/example/restapi/config/SecurityConfig.java (hunk 1 above)
Comment (locpnguyen2912) [reply src/main/java/com/example/restapi/config/SecurityConfig.java]: fixed by changing to define bean PasswordEncoder

File: src/main/java/com/example/restapi/config/SecurityConfig.java (hunk 2 above)
Comment (locpnguyen2912) [reply src/main/java/com/example/restapi/config/SecurityConfig.java]: fixed by changing to define bean PasswordEncoder

**Review by vokhanhhuy — APPROVED** [review]

**Pull request merged**